### DISCUSSION TITLE
interpolate cell data

### DISCUSSION THREAD

**Discussion by eino** [pyvista/pyvista#2219] [Q&A] [answered]
Hi,
I need to interpolate cell data from one polydata to another. But from what I've seen, `interpolate` only works with point data, not cell data. Is it so, or is there a way to make it transfer also cell data?

**Comment by eino** [comment, marked as answer]
To answer myself, this can be done quickly by running `cell_centers` first

**Reply by firstkingofrome** [reply]
Thank You that was helpful to me as well. I am just adding some test code that I needed to write to figure out what you meant in the hope that it helps someone else with the same issue:

(based on the interpolation example in the manual: https://docs.pyvista.org/examples/01-filter/interpolate.html)
```python
import pyvista as pv
from pyvista import examples

surface = examples.download_saddle_surface()
points = examples.download_sparse_points()
centers=surface.cell_centers()
interpolated = centers.interpolate(points, radius=12.0)
#now assign
surface['values']=interpolated['val']
surface.save("testIntepolate.vtk")
```

**Reply by MatthewFlamm** [reply]
Note that using `cell_centers` will result in a mesh that is smaller in physical size than the original mesh.  It may be better to use `cell_data_to_point_data`.  If you are using simulation data from CFD, it is usually better to get export point data directly to get the simulation point data rather than letting vtk use a generic method to determine.
